.github/workflows/maven-update-version.yml

```diff
@@ -0,0 +1,45 @@
+name: Update version with Maven
+# - validate that there is no dependency on snapshot
+# - update version to non-snapshot
+# - commit and tag
+# - update version to snapshot
+# - push changes
+
+on:
+  workflow_dispatch:
+    inputs:
+      update_type:
+        description: Update type
+        required: true
+        type: choice
+        options:
+          # `major` not yet supported
+          - minor
+          - bugfix
+
+jobs:
+  update-version:
+    name: Update version
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+          cache: maven
+      - name: Run Maven release:prepare
+        run: |
+          echo ${{ inputs.update_type }}
+          if [ ${{ inputs.update_type }} == "minor" ]; then
+            mvn -B -U -DdryRun=true release:prepare -DtagNameFormat=@{project.version} -DprojectVersionPolicyId=SemVerVersionPolicy
```

Comment:
I have added `-DdryRun=true` to fine-tune the action. Will be removed in a consecutive PR.